resources/testharness.js

```diff
@@ -1741,6 +1741,12 @@ policies and contribution forms [3].
             }
         };
 
+        if (self.Promise) {
+            this.done = new Promise(function(resolve) {
+                this_obj.doneResolve = resolve;
+            });
+        }
+
```

Comment:
@jgraham @gsnedders I've added this so developers can `await fetch_tests_from_worker(worker)`. This means:

* Promise tests can run in a predictable order.
* Service worker tests can complete before the next test unregisters it.

Reply:
@jgraham @gsnedders *nudge* I don't suppose one of you could take a look at this bit?

Reply:
These changes seem okay, but can you extract them into a separate PR and update the docs in `docs/_writing-tests/testharness-api.md`?

Reply:
@Ms2ger shall do. Thanks for taking a look.

Reply:
Extracted: https://github.com/w3c/web-platform-tests/pull/10978